ACCESS-OM3 Feedback on How to Run (#1043)
ACCESS-Hive/access-hive.github.io: docs/models/run-a-model/run-access-om3.md

The general layout of a _payu_ supported model run consists of two main directories:

- The _control_ directory contains the model configuration and serves as the execution directory for running the model (in this example, the cloned directory `~/access-om3/{{example_folder}}`).

> - The _laboratory_ directory, where all the model components reside. For {{ model }}, it is typically `/scratch/$PROJECT/$USER/access-om3`. Payu automatically creates this directory when a model configuration is run.

Author: Done in https://github.com/ACCESS-NRI/ACCESS-Hive-Docs/pull/1043/commits/2fc0547b0e423607e9d37bedb2277f7b4fa39043

By default, {{ model }} saves restart files after each run, allowing subsequent simulations to resume from a previously saved model state. The default {{ model }} run length and restart period can be changed (see [Change run length and restart period](#change-run-length-and-restart-period)).<br>
However, restart files can occupy significant disk space, and keeping all of them throughout an entire experiment is often not necessary. If disk space is limited, consider using _payu_'s restart files pruning feature, controlled by the `restart_freq` field of the `config.yaml`.

Every `restart_freq` _payu_ removes intermediate restart files, keeping only: 

- the two most recent restarts

> - restarts corresponding to the `restart_freq` interval

Reviewer: I would make it more explicit with AND.

```suggestion
Every `restart_freq` _payu_ removes intermediate restart files, keeping only the two most recent restarts and the restarts corresponding to the `restart_freq` interval.
```

Reviewer 2: I suggest rewording for improved readability to: 
_payu_ removes intermediate restart files every `restart_freq`, keeping only the two ...

- `T` &rarr; minute
- `S` &rarr; second

For example, to preserve the ability to restart {{ model }} every 5 model-years, set:
```yaml
restart_freq: '5YS'
```

If the run length was unchanged (e.g. 1 year) then each individal run (and therefore submission to the job queue) would be 1 model year.

> If `restart_freq` is 5 model years then when these restart files are no longer needed to continue an experiment, payu will discard 4 out of 5 of the restart files to save space.

Reviewer: I would not have an example with an unchanged value, but rather a changed one so it's more efficient. 

Also, I think an example where the run length `stop_n` is bigger than the `restart_freq` would make more sense. Otherwise I'm not getting the utility of the `restart_freq` field, since an experiment running for 1 year would never have 5 years of restart files.

I think in general this is very confused because I'm not sure how it plays with "continuing" an experiment.

What if I have a first run by setting `stop_n = 1 years`, `restart_n = 2 nmonths` and `restart_freq` = `6MS`: based on my understanding the model would run for 1 year, initially set to write restarts every 2 months (so at month number 2, 4, 6, 8, 10 and 12). But since `restart_freq` = `6MS`, after (or at the beginning of) every 6 months, it will prune the intermediate restart files keeping only the ones at every 6th month and the latest two restarts. Therefore, at month 6, it will prune the restarts by deleting restart 2 and keeping only 4 and 6, and at month 12 it will prune the restarts by deleting restart 4 and 8 and keeping 6, 10 and 12. 
Is that correct?

Author: I knew this would be a can of worms!

The example is trying to be, the experiment consists of at least 6 consecutive runs of 1 year each. Once the 6th run has happened, then the restarts for year 1-4  are deleted and only year 5 is kept.

Ill think about a reword / restructure / details on how consecutive runs form an experiment

Reviewer: What are the configuration settings to perform "6 consecutive runs of 1 year each"?

Reviewer: Yes, there is no mention of consecutive runs in the documentation and I think that would be useful for users to know. Especially if it's a common approach.

Reviewer 2: Good point, @atteggiani , the payu documentation explains this, but it would be good if it is mentioned here as well since it is a common approach.

For more information, check [_payu_ Configuration Settings documentation](https://payu.readthedocs.io/en/latest/config.html#model).

### Other configuration options

#### Model configuration {: .no-toc }

> This section tells _payu_ which driver to use for the main model configuration (`access-om3`) and the location of the model executable and all input files. These input files capture data needed for the experiment to be run, including grids, bathymetry, land/sea masks, initial conditions and atmospheric forcing data. There is some information on how these files are generated in the [{{ model }} configuration documentation]({{configs_docs}}).

Reviewer: ```suggestion
This section tells _payu_ which driver to use for the main model configuration (`access-om3`) and the location of the model executable and all input files. These input files capture data needed for the experiment to be run, including grids, bathymetry, land/sea masks, initial conditions and atmospheric forcing data. Information on how these files are generated can be found in the [{{ model }} configuration documentation]({{configs_docs}}).
```

    - /g/data/vk83/experiments/inputs/JRA-55/RYF/v1-4/data
```

> Other options for runtime configuration, including parameter choices and coupler configuration are within the text files in the `control` directory. See [Configuration Overview]({{configs_docs}}/configurations/Overview/) for more detail.

Author: I removed this sentence, 

line 330 now says: 

More details on model and _payu_ configuration are found in the 
[Configurations Overview](https://access-om3-configs.access-hive.org.au/configurations/Overview/) section of {{ model }} config docs.
